### PR TITLE
add ``ThreadPool`` JMX MXBean which exposes statistic for each pool

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: blackbox/docs/admin/monitoring.rst [hunk 1]
```diff
@@ -193,6 +193,80 @@ name and has the following attributes:
...
+
++---------------------+-----------------------------------------------------+
+| Name                | Description                                         |
++=====================+=====================================================+
+| ``poolSize``        | The current number of threads in the pool.          |
++---------------------+-----------------------------------------------------+
+| ``largestPoolSize`` | The largest number of threads that have ever        |
+|                     | simultaneously been in the pool.                    |
++---------------------+-----------------------------------------------------+
+| ``queueSize``       | The current number of threads in the queue.         |
```
Comment (mfussenegger) [inline blackbox/docs/admin/monitoring.rst]: I think this should be tasks instead of threads.

File: blackbox/docs/admin/monitoring.rst [hunk 2]
```diff
@@ -193,6 +193,80 @@ name and has the following attributes:
...
+It can be accessed using the ``io.crate.monitoring:type=ThreadPools`` object
+name and has following attributes:
+
++-----------------------+---------------------------------------------------------+
+| Name                  | Description                                             |
++=======================+=========================================================+
+| ``Generic``           | Thread pool statistic of the ``generic`` thread pool.   |
++-----------------------+---------------------------------------------------------+
+| ``Search``            | Thread pool statistics of the ``search`` thread pool    |
+|                       | used by read queries on user generated tables.          |
```
Comment (andreidan) [inline blackbox/docs/admin/monitoring.rst]: uber minor but "read queries" is a pleonasm. maybe we should only use `queries`?

File: blackbox/docs/admin/monitoring.rst [hunk 3]
```diff
@@ -193,6 +193,80 @@ name and has the following attributes:
...
++-----------------------+---------------------------------------------------------+
+| ``Bulk``              | Thread pool statistics of the ``bulk`` thread pool      |
+|                       | used for writing and deleting data.                     |
++-----------------------+---------------------------------------------------------+
+| ``Management``        | Thread pool statistics of the ``management`` thread     |
+|                       | pool used by management tasks like stats collecting,    |
+|                       | repository infos, allocations, etc.                     |
++-----------------------+---------------------------------------------------------+
+| ``Index``             | Thread pool statistics of the ``index`` thread pool     |
+|                       | used for writing blobs.                                 |
```
Comment (andreidan) [inline blackbox/docs/admin/monitoring.rst]: is this for blobs only?

File: blackbox/docs/admin/monitoring.rst (hunk 2 above)
Comment (seut) [reply blackbox/docs/admin/monitoring.rst]: would call it then read statements

File: blackbox/docs/admin/monitoring.rst [hunk 4]
```diff
@@ -193,6 +193,80 @@ name and has the following attributes:
...
++-----------------------+---------------------------------------------------------+
+| Name                  | Description                                             |
++=======================+=========================================================+
+| ``Generic``           | Thread pool statistic of the ``generic`` thread pool.   |
++-----------------------+---------------------------------------------------------+
+| ``Search``            | Thread pool statistics of the ``search`` thread pool    |
+|                       | used by read queries on user generated tables.          |
++-----------------------+---------------------------------------------------------+
+| ``Bulk``              | Thread pool statistics of the ``bulk`` thread pool      |
+|                       | used for writing and deleting data.                     |
```
[the remark this answers is not in the document]
Comment (seut) [reply blackbox/docs/admin/monitoring.rst]: it is also used for single writes/deletes

File: blackbox/docs/admin/monitoring.rst (hunk 3 above)
Comment (seut) [reply blackbox/docs/admin/monitoring.rst]: yes

File: enterprise/jmx-monitoring/src/main/java/io/crate/beans/ThreadPools.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+import static org.elasticsearch.threadpool.ThreadPool.Names.GENERIC;
+import static org.elasticsearch.threadpool.ThreadPool.Names.GET;
+import static org.elasticsearch.threadpool.ThreadPool.Names.INDEX;
+import static org.elasticsearch.threadpool.ThreadPool.Names.LISTENER;
+import static org.elasticsearch.threadpool.ThreadPool.Names.MANAGEMENT;
+import static org.elasticsearch.threadpool.ThreadPool.Names.REFRESH;
+import static org.elasticsearch.threadpool.ThreadPool.Names.SEARCH;
+import static org.elasticsearch.threadpool.ThreadPool.Names.SNAPSHOT;
+
+public class ThreadPools implements ThreadPoolsMXBean {
```
[the remark this answers is not in the document]
Comment (seut) [reply enterprise/jmx-monitoring/src/main/java/io/crate/beans/ThreadPools.java]: all mbeans use this naming pattern and i dont think we must repeat the package inside the class name

**Pull request merged**